[Concepts] Add Decls from the outer scope of the current lambda for conversion function constraints (#83420)
llvm/llvm-project: clang/docs/ReleaseNotes.rst

  (`#82258 <https://github.com/llvm/llvm-project/issues/82258>`_)
- Correctly immediate-escalate lambda conversion functions.
  (`#82258 <https://github.com/llvm/llvm-project/issues/82258>`_)
- Fixed a crash while checking constraints of a trailing requires-expression of a lambda, that the

>   expression references to an entity declared outside of the lambda. This is a reduction from

Reviewer: ```suggestion
  expression references to an entity declared outside of the lambda. Fixes
```

>   (`#64808 <https://github.com/llvm/llvm-project/issues/64808>`_).

Reviewer: ```suggestion
  (#GH64808)
```
